org.eclipse.jdt.core.manipulation/core extension/org/eclipse/jdt/internal/corext/fix/helper/WhileToForEach.java

```diff
@@ -418,19 +419,17 @@ public void rewrite(UseIteratorToForLoopFixCore upp, final WhileLoopToChangeHit
 			varBinding= type.resolveBinding();
 		}
 		if (type == null || varBinding == null) {
-			looptargettype= "java.lang.Object"; //$NON-NLS-1$
```

Comment:
Why did you remove this?  This is a fall back type.  In your new code, you set parametrizedType to null.  If the binding is null, you will set the type to null.  Bindings can come back null for many reasons so you need to be prepared.

Reply:
Sorry, I don't know why, but I must have overseen that `computeTypeArgument()` can return null.
I reverted that part.